fineract-loan/src/main/java/org/apache/fineract/portfolio/interestpauses/api/LoanInterestPauseApiResource.java

```diff
@@ -114,4 +117,38 @@ public List<InterestPauseResponseDto> retrieveInterestPausesByExternalId(
 
         return this.interestPauseReadPlatformService.retrieveInterestPauses(loanExternalId);
     }
+
+    @DELETE
+    @Path("/{loanId}/interest-pauses/{variationId}")
+    @Operation(summary = "Delete an interest pause period", description = "Deletes a specific interest pause period by its variation ID.")
+    @ApiResponses({ @ApiResponse(responseCode = "204", description = "No Content") })
+    public Response deleteInterestPause(@PathParam("loanId") @Parameter(description = "loanId") final Long loanId,
+            @PathParam("variationId") @Parameter(description = "variationId") final Long variationId) {
+
+        this.context.authenticatedUser().validateHasReadPermission(RESOURCE_NAME_FOR_PERMISSIONS);
+
+        final CommandWrapper commandRequest = new CommandWrapperBuilder().deleteInterestPause(loanId, variationId).build();
+
+        this.commandsSourceWritePlatformService.logCommandSource(commandRequest);
+
+        return Response.noContent().build();
+    }
+
+    @PUT
+    @Path("/{loanId}/interest-pauses/{variationId}")
+    @Consumes({ MediaType.APPLICATION_JSON })
+    @Produces({ MediaType.APPLICATION_JSON })
+    @Operation(summary = "Update an interest pause period", description = "Updates a specific interest pause period by its variation ID.")
+    @ApiResponses({ @ApiResponse(responseCode = "200", description = "OK") })
+    public CommandProcessingResult updateInterestPause(@PathParam("loanId") @Parameter(description = "loanId") final Long loanId,
+            @PathParam("variationId") @Parameter(description = "variationId") final Long variationId,
+            @RequestBody(required = true) final InterestPauseRequestDto request) {
+
+        this.context.authenticatedUser().validateHasReadPermission(RESOURCE_NAME_FOR_PERMISSIONS);
```

Comment:
I know i have missed this in the previous PR, but i noticed here... Just because the user has read permission on the loan, it shall not be able to add, modify or delete interest pauses! If we are validating permission that should be "UPDATE LOAN" in this case.